### PR TITLE
Refactor FlowNavigation

### PR DESCRIPTION
**Pull request by chohner**
Mostly replaces subflownavigation & subnavitem with nesting. This was done now to simplify the upcoming restyling

### CODE REVIEW

File: app/components/FlowNavigation.tsx [hunk 1]
```diff
@@ -44,45 +37,43 @@ const stateIsDone = (state: NavState) =>
...
-  navItems: NavItem[];
-  a11yLabels?: NavigationA11yLabels;
-}>) {
+  isChild = false,
+}: FlowNavigationProps & { isChild?: boolean }) => (
+  <ul
+    className={
+      isChild
+        ? "pt-8 pl-32 mr-8 min-w-fit max-w-fit md:min-w-[250px] md:max-w-[250px] break-words"
+        : ""
```
Comment (pgurusinga) [inline app/components/FlowNavigation.tsx:59]: maybe we could utilize classNames here

```suggestion
    className={
      classNames({ "pt-8 pl-32 mr-8 min-w-fit max-w-fit md:min-w-[250px] md:max-w-[250px] break-words": isChild })
```

File: app/components/FlowNavigation.tsx [hunk 2]
```diff
@@ -114,39 +106,48 @@ function NavItem({
...
+        isDisabled,
+    },
   );
   const iconId = useId();
 
   return (
     <li
-      key={destination}
-      className={"list-none border-t-2 border-white first:border-t-0"}
+      className={`list-none ${isChild ? "" : "border-t-2 border-white first:border-t-0"}`}
```
Comment (pgurusinga) [inline app/components/FlowNavigation.tsx:121]: is it possible to do it with className and negate the value?

```suggestion
      className={`list-none classNames({ "border-t-2 border-white first:border-t-0": !isChild })`}
```

File: app/components/FlowNavigation.tsx [hunk 3]
```diff
@@ -44,45 +37,43 @@ const stateIsDone = (state: NavState) =>
...
   itemFinished: string;
   itemOpen: string;
 };
+type FlowNavigationProps = Readonly<{
+  navItems: NavItem[];
+  a11yLabels?: NavigationA11yLabels;
+}>;
 
-export default function FlowNavigation({
+const NavigationList = ({
```
Comment (pgurusinga) [inline app/components/FlowNavigation.tsx:50]: could you please add component testing for `FlowNavigation`?

File: app/components/FlowNavigation.tsx (hunk 1 above)
Comment (chohner) [reply app/components/FlowNavigation.tsx:59]: Already done in https://github.com/digitalservicebund/a2j-rechtsantragstelle/pull/856/ :)

File: app/components/FlowNavigation.tsx (hunk 2 above)
Comment (chohner) [reply app/components/FlowNavigation.tsx:121]: see above

**Review by pgurusinga — APPROVED** [review]

**Pull request merged**